src/pages/RepoPage/context.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, useContext, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+type Breadcrumb = {
+  pageName: string
+  text?: string
+  children?: JSX.Element
+  readOnly?: boolean
+}
+
+const RepoBreadcrumbContext = createContext<{
+  setBaseCrumbs: (crumbs: Breadcrumb[]) => void
+  breadcrumbs: Breadcrumb[]
+  setBreadcrumbs: (crumbs: Breadcrumb[]) => void
+}>({
```

Comment:
`l`: Thoughts on pulling this out into it's own type, may also be able to import it incase anyone needs the type if there's something funky going on with the context?

Reply:
yea good idea!